### PR TITLE
#52: Add interactive questions for each step (closes #52)

### LINKED ISSUE #52: Add interactive questions for each step
It would be nice if smooth-release paused with a `y/n` prompt before each step. In this way a new user of smooth-release would be guided through the steps and could double check the tool is not doing anything he/she is not expecting.

It should be possible to disable the interactive mode via the `.smooth-releaserc` config file.

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Issue #52

### tests performed
- run `./smooth-release`
  - validations run and fails (I'm not on `master`)
- run `./smooth-release --no-validations`
  - skips validations and prompts for every other task. skips also "gh-release-all"
- add "tasks" map to `.smooth-releaserc`, set "validations" to false and run `'./smooth-release`
  - same result as previous test ✅ 

[https://gyazo.com/a906bd7d657f928d69ccdd0e30d35668](https://gyazo.com/a906bd7d657f928d69ccdd0e30d35668)

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.}

### CODE REVIEW

**Comment by FrancescoCioria** [comment]
@dhinus what should be default when the user taps "Enter" key in your opinion? at the moment is set to "n".

**Comment by dhinus** [comment]
@FrancescoCioria not sure, I think "n" is fine. Ideally I would like it to perform the task after pressing "y", then stop for the next question, instead of asking all the questions first. Not a big issue though.

How hard would it be to add a "preview" of the task, showing the version number and the changelog? It can probably be done in a separate pull request.

File: src/index.js [hunk 1]
```diff
@@ -1,45 +1,65 @@
...
+      await version(mainArgument);
+    }
 
-    argv.changelog && await changelog();
+    if (await promptUserBeforeRunningTask('changelog', 'Do you want to run "changelog" task and update the CHANGELOG.md file?')) {
+      await changelog();
+    }
 
-    argv['gh-release'] && await release({ all: false });
+    if (await promptUserBeforeRunningTask('gh-release', 'Do you want to run "gh-release" task and create a release on GitHub for the last version of you library?')) {
```
Comment (dhinus) [inline src/index.js]: I think this should read
> to run **the** "gh-release" task

or
> to run **the task** "gh-release"

(not sure which one is better)

Comment (dhinus) [reply src/index.js]: Also,
> you**r** library

Comment (FrancescoCioria) [reply src/index.js]: 👍

File: README.md [hunk 2]
```diff
@@ -130,10 +108,40 @@ smooth-release --changelog --gh-release-all # first time using smooth-release on
...
-If you want to change parts of it you can define a JSON config file in the root directory of your project named `.smooth-releaserc`.
+setting a task to `null` means that `smooth-release` will prompt you every time before running the task:
+![image](https://cloud.githubusercontent.com/assets/4029499/21606902/e78f23d0-d1b2-11e6-9c17-b4bccf853856.png)
+
+If you want to change parts of the default config you can define a JSON file in the root directory of your project named `.smooth-releaserc`.
 
 The file will be recursively merged into `defaultConfig` (NB: arrays are replaced, not merged!).
+
+
+## CLI arguments
```
Comment (dhinus) [inline README.md:131]: Can we specify in the Readme that by using any CLI argument, all prompts are disabled?

**Comment by FrancescoCioria** [comment]
> Ideally I would like it to perform the task after pressing "y", then stop for the next question, instead of asking all the questions first.

This is exactly how it works, in the GIF you don't notice it because i press "Enter" which defaults to "n". If you press "y" it runs that task and, after the task has finished, it prompts you for the following one.

File: README.md (hunk 2 above)
Comment (FrancescoCioria) [reply README.md:131]: Isn't [this](https://github.com/FrancescoCioria/smooth-release/pull/56/files#diff-04c6e90faac2675aa89e2176d2eec7d8R147) enough? You think we should re-write it more clearly?

Comment (dhinus) [reply README.md:131]: Maybe adding this at the end of the section?
> If you specify one ore more negative argument, interactive prompts will be displayed for the remaining arguments.
> If you specify one or more positive argument, all interactive prompts will be disabled and only the whitelisted tasks will be run.

File: src/index.js (hunk 1 above)
Comment (dhinus) [reply src/index.js]: Don't forget "you library" -> "your library" (lines 44 and 52)

File: README.md (hunk 2 above)
Comment (FrancescoCioria) [reply README.md:131]: Ok, thank you!

**Pull request merged**